src/Bridges/ApplicationDI/ApplicationExtension.php

```diff
@@ -65,4 +75,67 @@ public function loadConfiguration()
 		}
 	}
 
+
+	public function beforeCompile()
+	{
+		$config = $this->validateConfig($this->defaults);
+		$container = $this->getContainerBuilder();
+		$counter = 0;
+
+		foreach ($this->findClasses() as $class) {
+			if ($services = $container->findByType($class, FALSE)) {
```

Comment:
Nope...
a) It goes over the list compiled by `findClasses()`, thus ignoring presenters manually registered as services by user.
b) There is still the problem I mentioned earlier - if `findClasses()` returns the list containing `BPresenter` before `APresenter`, you'll get only `BPresenter` registered with a wrong tag `nette.presenter=APresenter`. When the loop later gets to processing `APresenter`, the `findByType("APresenter", FALSE)` will return previously registered definition of `BPresenter` and set it wrong tag.